addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -123,211 +144,173 @@ public void run() {
-            logger.debug("BRIDGE:    Port HTTP(S)        {}", getBridgeConfig().getHttpPort());
-            logger.debug("BRIDGE:    Port Telnet         {}", getBridgeConfig().getTelnetPort());
-            logger.debug("BRIDGE:    Server Path         {}", getBridgeConfig().getServerBasePath());
-            logger.debug("BRIDGE:    User:               {}", getBridgeConfig().getUserName());
-            logger.debug("BRIDGE:    Refresh interval:   {}", getBridgeConfig().getRefreshInterval());
-            logger.debug("BRIDGE:    Low  prio. refresh: {}", getBridgeConfig().getRefreshIntervalLowPriorityTask());
-            logger.debug("BRIDGE:    Autodiscovery:      {}", getBridgeConfig().getAutodiscoverThings());
-
-            closeConnection();
-
-            zoneMinderConnection = ZoneMinderFactory.CreateConnection(getBridgeConfig().getProtocol(),
-                    getBridgeConfig().getHostName(), getBridgeConfig().getHttpPort(), getBridgeConfig().getTelnetPort(),
-                    getBridgeConfig().getServerBasePath(), getBridgeConfig().getUserName(),
-                    getBridgeConfig().getPassword(), 3000);
+            zoneMinderConnection = null;
 
             taskRefreshData = null;
-            taskPriorityRefreshData = null;
 
-        } catch (Exception ex) {
-            logger.error("[BRIDGE]: 'ZoneMinderServerBridgeHandler' failed to initialize. Exception='{}'",
-                    ex.getMessage());
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
-        } finally {
+            updateStatus(ThingStatus.OFFLINE);
+
+            ZoneMinderBridgeServerConfig config = getBridgeConfig();
+            logger.info("{}: ZoneMinder Server Bridge Handler Initialized", getLogIdentifier());
+            logger.debug("{}:    HostName:           {}", getLogIdentifier(), config.getHost());
+            logger.debug("{}:    Protocol:           {}", getLogIdentifier(), config.getProtocol());
+            logger.debug("{}:    Port HTTP(S)        {}", getLogIdentifier(), config.getHttpPort());
+            logger.debug("{}:    Port Telnet         {}", getLogIdentifier(), config.getTelnetPort());
+            logger.debug("{}:    Portal Path         {}", getLogIdentifier(), config.getServerBasePath());
+            logger.debug("{}:    API Path            {}", getLogIdentifier(), config.getServerApiPath());
+            logger.debug("{}:    Refresh interval:   {}", getLogIdentifier(), config.getRefreshIntervalNormal());
+            logger.debug("{}:    Low  prio. refresh: {}", getLogIdentifier(), config.getRefreshIntervalLow());
+            logger.debug("{}:    Autodiscovery:      {}", getLogIdentifier(), config.getAutodiscoverThings());
+
             startWatchDogTask();
-            isInitialized = true;
-        }
-    }
+            initialized = true;
+            return;
+        } catch (Exception ex) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
Seems to be there onbly for historic reasons. Will be removed :-)